### PR TITLE
Refactor and tests

### PR DESCRIPTION
**Pull request by knopki**
Move to `TypedDict` as DTOs. Implement all API endpoints. Implements tests.

### CODE REVIEW

**Comment by knopki** [comment]
Obviously, the tests still need to be fixed :(

File: README.md [hunk 1]
```diff
@@ -25,28 +26,35 @@ async def main():
     async with MetisAPIAsync(API_URL, auth=MetisTokenAuth("admin@test.com")) as client:
         print(await client.v0.auth.whoami())
         data = await client.v0.datasources.create(content)
-        calc = await client.v0.calculations.create(data.id)
+        calc = await client.v0.calculations.create(data["id"])
```
Comment (blokhin) [inline README.md:29]: @knopki is there a specific reason over this syntax change?

**Review by blokhin — APPROVED** [review]

**Comment by blokhin** [comment]
@knopki can it go to PyPI now?

**Comment by knopki** [comment]
> @knopki can it go to PyPI now?

Yes. There are some problems on python3.8, but I think it's just buggy tests.

**Pull request merged**